osi_vehicle.proto

```diff
@@ -0,0 +1,692 @@
+        }
+    }
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the steering wheel.
+    //
+    message VehicleSteeringWheel
+    {
+        // Angle, angle-speed and torque.
+        // See osi_common_extension.
+        //
+        optional SteeringWheel steering_wheel = 1;
+
+        // Spring-stiffness of the steering in Nm/deg.
+        //
+        // Unit: [N*m/deg]
+        //
+        optional double stw_springstiffness = 2;
+
+        // Damping of the steering in Nm*s/deg.
+        //
+        // Unit: [N*m*s/deg]
+        //
+        optional double stw_damping = 3;
+
+        // Friction of the steering in Nm.
+        //
+        // Unit: [N*m]
+        //
+        optional double stw_friction = 4;
+
+        // Hands Off Detection.
+        //
+        optional bool are_hands_off = 5;
+    }
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the wheels.
+    // It is made usage of the wheel-message to shorten the code.
+    //
+    message VehicleWheels
+    {
+        // Contains the physical description of the front-left wheel.
+        //
+        optional Wheel wheel_front_left = 1;
+
+        // Contains the physical description of the front-right wheel.
+        //
+        optional Wheel wheel_front_right = 2;
+
+        // Contains the physical description of the rear-left wheel.
+        //
+        optional Wheel wheel_rear_left = 3;
+
+        // Contains the physical description of the rear-right wheel.
+        //
+        optional Wheel wheel_rear_right = 4;
+    }
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the physical description of a wheel.
+    //
+    message Wheel
+    {
+        // Contains the friction-coefficient of each wheel.
+        // Dimensionless.
+        //
+        // Unit: []
+        //
+        optional double friction_coefficient = 1;
+
+        // Contains the x, y and z-coordinate of the contact point of 
+        // the wheel, so that walking, torsion and deflation can be visualized.
+        // relative to the center of the wheel.
+        //
+        // Unit: [m]
+        //
+        optional Vector3d contact_point = 2;
+
+        // Contains the rotational speed of each wheel per second.
+        //
+        // Unit: [rad/s]
+        //
+        optional double rotational_speed = 3;
+
+        // Contains the steering angle of each wheel.
+        //
+        // Unit: [rad]
+        //
+        optional double steeringangle = 4;
+
+        // Contains the camber of each wheel.
+        // 
+        // Negative camber if the bottom of the wheel is farther out than the
+        // top.
+        // For more information: https://en.wikipedia.org/wiki/Camber_angle.
+        //
+        // Unit: [rad]
+        //
+        optional double camber = 5;
+
+        // Contains the tirepressure of each tire.
+        //
+        // Unit: [Pa]
```

Comment:
I there the need to have a temperature reference here?

Reply:
It can be helpful.